### ISSUE TITLE
Lua's varargs not visible inside quote

### ISSUE THREAD

**Issue by manopapad** [zdevito/terra#284]
I'm referring to the `...` special variable that Lua uses to support variable-argument functions. In the reproducer below, storing the varargs in a table first and then using them works fine, but trying to build the table inside the quote doesn't work (the `...` seems to be empty at that point).

```
local C = terralib.includec('stdio.h')

local function mkPrintQuoteBad(msg, ...)
  return quote
    C.printf(msg, [{...}])
  end
end

local function mkPrintQuoteGood(msg, ...)
  local args = {...}
  return quote
    C.printf(msg, [args])
  end
end

local terra bar(x : int, y : int)
  [mkPrintQuoteGood('Called with: x=%d, y=%d\n', x, y)]
  [mkPrintQuoteBad('Called with: x=%d, y=%d\n', x, y)]
end

bar:printpretty()
bar(1,2)
```

**Comment by Qix-** [comment]
Yeah this is blocking some very useful use-cases, unfortunately

**Comment by elliottslaughter** [comment]
@Qix- What's blocked for you? If I understand the issue, this is a purely cosmetic problem, since you can always dump the contents of `...` into a local variable.

**Comment by Qix-** [comment]
Apologies, my issue is different - using varargs in a `terra` function directly doesn't seem to work at all and requires a macro or lua function. However, the following compiles just fine.

```terra
terra some_function(fmt: &int8, ...)
    -- ...
end
```

As well, since Terra doesn't import non-trivial macros, `stdarg.h` won't work here either. So using the `...` in terra functions is a bit strange.

**Comment by elliottslaughter** [comment]
Yeah, I'm not aware of `...` being supported in Terra functions, I'd guess it's just by chance that the compiler doesn't throw an error there.
